GSA-TTS/FAC
# 1904 Change general form's "state" fields to dropdown

closes https://github.com/GSA-TTS/FAC/issues/1904

Pulls states out into their own JSON document, which is now referenced both by the schemas and the frontend form generation code.

To test locally:
* Pull this branch.
* `make docker-clean && make docker-first-run && docker compose up`
* Open the app in your browser and work through the new audit flow until you get to the general info form.
* Under the "Auditee details" heading, try selecting a state from the dropdown menu.
* Under the "Audit firm/organization address" heading, try typing a two-letter state abbreviation in the dropdown menu. 
* Complete the form and submit.
* Navigate back to the general info form you just submitted and verify that the states you selected are still there.

## PR checklist: submitters


## PR checklist: reviewers

        
The larger the PR, the stricter we should be about these points.
